packages/amplify_datastore/ios/Classes/DataStoreHubEventStreamHandler.swift

```diff
@@ -0,0 +1,140 @@
+                        }
+                    case HubPayload.EventName.DataStore.subscriptionsEstablished :
+                        do {
+                            let subscriptionsEstablished =  try FlutterSubscriptionsEstablishedEvent(payload: payload)
+                            self.sendEvent(flutterEvent: subscriptionsEstablished.toValueMap())
+                        } catch {
+                            self.sendError(description: castingError, eventName: "subscriptionsEstablished")
+                        }
+                    case HubPayload.EventName.DataStore.syncQueriesStarted :
+                        do {
+                            let syncQueriesStarted =  try FlutterSyncQueriesStartedEvent(payload: payload)
+                            self.sendEvent(flutterEvent: syncQueriesStarted.toValueMap())
+                        } catch {
+                            self.sendError(description: castingError, eventName: "syncQueriesStarted")
+                        }
+                    case HubPayload.EventName.DataStore.modelSynced :
+                        do {
+                            let modelSynced =  try FlutterModelSyncedEvent(payload: payload)
+                            self.sendEvent(flutterEvent: modelSynced.toValueMap())
+                        } catch {
+                            self.sendError(description: castingError, eventName: "modelSynced")
+                        }
+                    case HubPayload.EventName.DataStore.syncQueriesReady :
+                        do {
+                            let syncQueriesReady =  try FlutterSyncQueriesReadyEvent(payload: payload)
+                            self.sendEvent(flutterEvent: syncQueriesReady.toValueMap())
+                        } catch {
+                            self.sendError(description: castingError, eventName: "syncQueriesReady")
+                        }
+                    case HubPayload.EventName.DataStore.ready :
+                        do {
+                            let ready =  try FlutterReadyEvent(payload: payload)
+                            self.sendEvent(flutterEvent: ready.toValueMap())
+                        } catch {
+                            self.sendError(description: castingError, eventName: "ready")
+                        }
+                    case HubPayload.EventName.DataStore.outboxMutationEnqueued :
+                        do {
+                            guard let outboxMutationEnqueued = payload.data as? OutboxMutationEvent else {
+                                      throw FlutterDataStoreError.hubEventCast
+                                  }
+                            let modelName = outboxMutationEnqueued.modelName
+                            let schema = try FlutterDataStoreRequestUtils.getModelSchema(modelSchemas: self.flutterModelRegistration?.modelSchemas ?? [:], modelName: modelName)
```

Comment:
This will result in a decodingError from `FlutterDataStoreRequestUtils.getModelSchema` and we are losing this context since we won't see this error in Dart.

Reply:
I changed to this:
```
case HubPayload.EventName.DataStore.outboxMutationProcessed :
                        do {
                             guard let outboxMutationProcessed = payload.data as? OutboxMutationEvent else {
                                      throw FlutterDataStoreError.hubEventCast
                                  }
                            let modelName = outboxMutationProcessed.modelName
                            var schema: ModelSchema?
                            do {
                                schema = try self.getSchema(modelName: "TEST")
                            } catch {
                                print("Error getting model schema for \(modelName ) model: \(error)")
                            }
                            let flutterOutboxMutationProcessed = try FlutterOutboxMutationProcessedEvent(
                                outboxMutationProcessed: outboxMutationProcessed,
                                eventName: payload.eventName,
                                schema: schema)
                            self.sendEvent(flutterEvent: flutterOutboxMutationProcessed.toValueMap())
                        } catch {
                            self.sendError(hubEvent: shortEventName(eventName: payload.eventName))
                        }

```